File: MetaGeta/MetaGeta.GUI/ViewModels/NullViewModel.vb
Public Class NullViewModel
    Inherits NavigationTab

    Private ReadOnly m_Caption As String
    Private ReadOnly m_Icon As System.Windows.Media.ImageSource

    Public Sub New(ByVal caption As String, ByVal icon As ImageSource)
        m_Caption = caption
        m_Icon = icon
    End Sub

    Public Overrides ReadOnly Property Caption() As String
        Get
            Return m_Caption
        End Get
    End Property

    Public Overrides ReadOnly Property Icon() As System.Windows.Media.ImageSource
        Get
            Return m_Icon
        End Get
    End Property
End Class

File: MetaGeta/MetaGeta.GUI/ViewModels/NullViewModel.cs
// Copyright 2009 Will Thomas
// 
// This file is part of MetaGeta.
// 
// MetaGeta is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
// 
// MetaGeta is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
// GNU General Public License for more details.
// 
// You should have received a copy of the GNU General Public License
// along with MetaGeta. If not, see <http://www.gnu.org/licenses/>.

#region

using System.Windows.Media;

#endregion

namespace MetaGeta.GUI {
    public class NullViewModel : NavigationTab {
        private readonly string m_Caption;

        private readonly ImageSource m_Icon;

        public NullViewModel(string caption, ImageSource icon) {
            m_Caption = caption;
            m_Icon = icon;
        }

        public override string Caption {
            get { return m_Caption; }
        }

        public override ImageSource Icon {
            get { return m_Icon; }
        }
    }
}